get_started/simple_pipeline.md

```diff
@@ -0,0 +1,247 @@
+    output_caps: %Membrane.Caps.Audio.Raw{
+	  format: :s16le,
+	  sample_rate: 48000,
+	  channels: 2
+	}
+  },
+  portaudio: Membrane.PortAudio.Sink
+}
+```
+
+The keys in that keyword list are just a names we gave to elements. We're going to need them when linking.
+
+#### Linking elements
+
+Now we should link them in proper order. Each membrane element is one of three types: Source, Sink or Filter. The main difference is that Source provides only output pads, Sink only input and Filter both input and output. That means Source element can only start pipelines (it's not prepared to receive any data from other elements), Sink can only end pipeline (it will not send any data to succeeding elements), and Filters can be in the middle (it will receive, process and send data further). In our case a links declaration will look like that: 
+
+```elixir
+links = [
+  link(:file)
+  |> to(:decoder)
+  |> to(:converter)
+  |> to(:portaudio)
+]
+```
+
+File Source read bytes from our mp3, sends them to decoder. Decoder, after decoding, sends them to converter. Converter, after conversion sends them to sink. Portaudio sink receives them and plays music through Portaudio 🎶
+
+#### Parent Spec
+
+Tha last but not least is to take elements and links together into a proper structure:
+
+A structure here is `Membrane.ParentSpec` [docs](https://hexdocs.pm/membrane_core/Membrane.ParentSpec.html). You can also declare here other options if needed. In our pipeline `ParentSpec` will contains only children elements and links between them:
+
+```elixir
+spec = %ParentSpec{children: children, links: links}
+```
+
+At the and we need to return a proper tuple from `handle_init`. We can choose  from options described [here](https://hexdocs.pm/membrane_core/Membrane.Pipeline.html#t:callback_return_t/0-return-values). They are common for all callbacks, but as we're initialising pipeline we need to choose a one wich declares [actions to take with the pipeline](https://hexdocs.pm/membrane_core/Membrane.Pipeline.Action.html). Our action will be [`spec`](https://hexdocs.pm/membrane_core/Membrane.Pipeline.Action.html#t:spec_t/0):
+
+```elixir
+{{:ok, spec: spec}, %{}}
+```
+We're passing empty map for state, as we don't need anything to be stored as state.
+
+After that a full modul will look like that:
+
+```elixir
+defmodule Hello do
+
+  use Membrane.Pipeline
+
+  @impl true
+  def handle_init(path_to_mp3) do
+	children = %{
+	  file: %Membrane.File.Source{location: path_to_mp3},
+	  decoder: Membrane.MP3.MAD.Decoder,
+	  converter: %Membrane.FFmpeg.SWResample.Converter{
+	    output_caps: %Membrane.Caps.Audio.Raw{
+		  format: :s16le,
+		  sample_rate: 48000,
+		  channels: 2
+		}
+	  },
+	  portaudio: Membrane.PortAudio.Sink
+    }
+
+	links = [
+	  link(:file)
+	  |> to(:decoder)
+	  |> to(:converter)
+	  |> to(:portaudio)
+	]
+
+	spec = %ParentSpec{children: children, links: links}
+
+	{{:ok, spec: spec}, %{}}
+  end
+end
+```
```

Comment:
I am just wondering - Do we want to once again provide the full code snippet? Or perhaps it would be better not to put the whole snippet on the top of the tutorial as it will be put there?

Reply:
Yes, the first one is just a demonstration of how it will look, the last one is the result. I know it's repetition but it's worth in my opinion to repeat it, as it's first usage example of Membrane. We'll skip the demonstration in next tutorials.